### ISSUE TITLE
augmentation.mix: RandomCutMixV2(num_mix>1, same_on_batch=True) repeats one cut but labels every mix

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4805]
## Summary

With `same_on_batch=True`, `RandomCutMixV2(num_mix=k)` gives every mix the same pairing and the same cut geometry. In the image, each mix after the first repeats the cut of the first: same region, same donor. The class output still lists `k` mixes, each crediting that donor with the full cut weight. For `num_mix=2`, the donor holds 39% of the pixels here, but it is labelled twice with `lambda = 0.609` (credit `2 * 0.391`), while the original label also appears twice.

The pairing is drawn as one `(num_mix, B)` sample with `same_on_batch=True` (`random_generator/_2d/cutmix.py`, `_adapted_sampling((self.num_mix, batch_size), ...)`), so the rows of that sample are identical. With `same_on_batch=False` the two mixes pair identically only by chance (2 of 50 seeds).

The docstring says `same_on_batch=True` "shares one geometry" across rows and mixes, and that part is intended. The labels contradicting the pixels is not.

## Repro

```python
import torch, kornia, kornia.augmentation as K
print("kornia", kornia.__version__, "torch", torch.__version__)
torch.manual_seed(0)
image = torch.arange(4.0).view(4, 1, 1, 1).expand(4, 1, 8, 8).clone()  # row i is filled with the value i
labels = torch.tensor([0, 1, 2, 3])
aug = K.RandomCutMixV2(num_mix=2, same_on_batch=True, p=1.0, use_correct_lambda=True, data_keys=["input", "class"])
out, mixed = aug(image, labels)
print("mix_pairs per mix:", aug._params["mix_pairs"].tolist())
print("labels, mix 0:", mixed[0].tolist())
print("labels, mix 1:", mixed[1].tolist())
for i in range(4):
    print(f"row {i}: values {out[i, 0].unique().tolist()}, fraction replaced {(out[i, 0] != i).float().mean().item():.4f}")
same = 0
for seed in range(50):
    torch.manual_seed(seed)
    p = K.RandomCutMixV2(num_mix=2, same_on_batch=True, p=1.0, use_correct_lambda=True).forward_parameters(torch.Size([4, 1, 8, 8]))
    same += bool((p["mix_pairs"][0] == p["mix_pairs"][1]).all())
print(f"identical pairing in both mixes: {same}/50 seeds (2/50 with same_on_batch=False)")
```

```
kornia 0.9.0rc1 torch 2.14.0
mix_pairs per mix: [[1, 2, 3, 0], [1, 2, 3, 0]]
labels, mix 0: [[0.0, 1.0, 0.609375], [1.0, 2.0, 0.609375], [2.0, 3.0, 0.609375], [3.0, 0.0, 0.609375]]
labels, mix 1: [[0.0, 1.0, 0.609375], [1.0, 2.0, 0.609375], [2.0, 3.0, 0.609375], [3.0, 0.0, 0.609375]]
row 0: values [0.0, 1.0], fraction replaced 0.3906
row 1: values [1.0, 2.0], fraction replaced 0.3906
row 2: values [2.0, 3.0], fraction replaced 0.3906
row 3: values [0.0, 3.0], fraction replaced 0.3906
identical pairing in both mixes: 50/50 seeds (2/50 with same_on_batch=False)
```

## Possible fixes

- Collapse the mixes when `same_on_batch=True` and `num_mix > 1`: emit one mix, or raise, since extra identical cuts add nothing.
- Or share the geometry across rows only, and draw an independent pairing and placement per mix, so each mix is a real additional cut.

Found while reviewing #4791, which documents this as a wart on `RandomCutMixV2`.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
